### PR TITLE
#17748 fix jump password when not providing password

### LINKED ISSUE #17748: SSH Auth Fail after 22.2.1 update
### Description

Updating to 22.2.1 breaks SSH+Jump server with public key connection. Downgrading to 22.2.0 makes it work again.
Error says "Auth fail", log attached.
[dbeaver.log](https://github.com/dbeaver/dbeaver/files/9658583/dbeaver.log)

### DBeaver Version

Community Edition 22.2.1

### Operating System

Windows 10

### Database and driver

_No response_

### Steps to reproduce

1. Configure SSH connection through a jump server with pulic key auth on both
2. Test tunnel, or just connect to database
3. Auth fail

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Destrolaric**
Closes #17748
Now unticking the save password checkbox also asks for the password for jump server configuration if it's enabled. Also, now for public keys password dialog will show up only if the key is encrypted.

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.net.ssh.ui/src/org/jkiss/dbeaver/ui/net/ssh/SSHTunnelConfiguratorUI.java [hunk 1]
```diff
@@ -285,15 +295,41 @@ protected void execute(@NotNull DBRProgressMonitor monitor) throws Throwable {
...
+                        DBPAuthInfo dbpAuthInfo = promptCredentialsDialog(authType, configuration.getUserName(),
+                            configuration.getPassword());
                         if (dbpAuthInfo != null) {
                             if (authType.equals(SSHConstants.AuthType.PASSWORD)) {
                                 configuration.setUserName(dbpAuthInfo.getUserName());
                             }
                             configuration.setPassword(dbpAuthInfo.getUserPassword());
                         }
                     }
+                    if (configuration.getBooleanProperty(getJumpServerSettingsPrefix() + "enabled")) { //$NON-NLS-1$
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.net.ssh.ui/src/org/jkiss/dbeaver/ui/net/ssh/SSHTunnelConfiguratorUI.java]: What this code does? We already load secure props, arent's we?

File: plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceDescriptor.java [hunk 2]
```diff
@@ -1042,6 +1047,47 @@ public boolean connect(DBRProgressMonitor monitor, boolean initialize, boolean r
         }
     }
 
+    private boolean askForSSHJumpServerPassword(@NotNull DBWHandlerConfiguration tunnelConfiguration) {
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceDescriptor.java:1050]: Don't use hardcoded strings, use constants (password, etc)

File: plugins/org.jkiss.dbeaver.net.ssh.ui/src/org/jkiss/dbeaver/ui/net/ssh/SSHTunnelConfiguratorUI.java (hunk 1 above)
Comment (Destrolaric) [reply plugins/org.jkiss.dbeaver.net.ssh.ui/src/org/jkiss/dbeaver/ui/net/ssh/SSHTunnelConfiguratorUI.java]: Checks if the jump server is enabled, if it is we probably should summon a password dialog for it. Then writes new creds for jump server

**Comment by Matvey16** [comment]
NPE for PKCS8 keys

```
java.lang.NullPointerException
	at com.jcraft.jsch.KeyPairPKCS8.getPublicKeyBlob(KeyPairPKCS8.java:199)
	at com.jcraft.jsch.IdentityFile.getPublicKeyBlob(IdentityFile.java:71)
	at com.jcraft.jsch.LocalIdentityRepository.add(LocalIdentityRepository.java:63)
	at com.jcraft.jsch.JSch.addIdentity(JSch.java:470)
	at com.jcraft.jsch.JSch.addIdentity(JSch.java:407)
	at com.jcraft.jsch.JSch.addIdentity(JSch.java:366)
	at org.jkiss.dbeaver.model.net.ssh.SSHUtils.isKeyFileEncrypted(SSHUtils.java:66)
	at org.jkiss.dbeaver.model.net.ssh.SSHTunnelImpl.getRequiredCredentials(SSHTunnelImpl.java:130)
	at org.jkiss.dbeaver.ui.net.ssh.SSHTunnelConfiguratorUI$8.execute(SSHTunnelConfiguratorUI.java:299)
	at org.jkiss.dbeaver.ui.net.ssh.SSHTunnelConfiguratorUI$TunnelConnectionTestJob.run(SSHTunnelConfiguratorUI.java:733)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:105)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
```

**Comment by Destrolaric** [comment]
NPE is present in version 21.1.4. We are opening a separate ticket for this issue.

**Comment by Matvey16** [comment]
Verified

**Pull request merged**